test/hotspot/jtreg/runtime/Thread/SuspendAtExit.java

```diff
@@ -52,59 +60,122 @@ public void run() {
+        try {
+            System.loadLibrary(AGENT_LIB);
+            log("Loaded library: " + AGENT_LIB);
+        } catch (UnsatisfiedLinkError ule) {
+            log("Failed to load library: " + AGENT_LIB);
+            log("java.library.path: " + System.getProperty("java.library.path"));
+            throw ule;
+        }
 
-        for (int i = 0; i < N_THREADS; i++ ) {
-            threads[i] = new SuspendAtExit();
-            int late_count = 1;
-            threads[i].start();
+        int timeMax = 0;
+        if (args.length == 0) {
+            timeMax = DEF_TIME_MAX;
+        } else {
             try {
-                // Wait for the worker thread to get going.
-                threads[i].startSyncObj.await();
-
-                // This interrupt() call will break the worker out
-                // of the exitSyncObj.await() call and the suspend()
-                // calls will come in during thread exit.
-                threads[i].interrupt();
-                for (; late_count <= N_LATE_CALLS; late_count++) {
-                    threads[i].suspend();
-
-                    if (!threads[i].isAlive()) {
-                        // Done with Thread.suspend() calls since
-                        // thread is not alive.
-                        break;
-                    }
-                    threads[i].resume();
-                }
-            } catch (InterruptedException e) {
-                throw new Error("Unexpected: " + e);
+                timeMax = Integer.parseUnsignedInt(args[0]);
+            } catch (NumberFormatException nfe) {
+                System.err.println("'" + args[0] + "': invalid timeMax value.");
+                    usage();
             }
+        }
 
-            System.out.println("INFO: thread #" + i + ": made " + late_count +
-                               " late calls to java.lang.Thread.suspend()");
-            System.out.println("INFO: thread #" + i + ": N_LATE_CALLS==" +
-                               N_LATE_CALLS + " value is " +
-                               ((late_count >= N_LATE_CALLS) ? "NOT " : "") +
-                               "large enough to cause a Thread.suspend() " +
-                               "call after thread exit.");
+        System.out.println("About to execute for " + timeMax + " seconds.");
 
-            try {
-                threads[i].join();
-            } catch (InterruptedException e) {
-                throw new Error("Unexpected: " + e);
-            }
-            threads[i].suspend();
-            threads[i].resume();
-            if (threads[i].isAlive()) {
-                throw new Error("Expected !Thread.isAlive() after thread #" +
-                                i + " has been join()'ed");
+        long count = 0;
+        long start_time = System.currentTimeMillis();
+        while (System.currentTimeMillis() < start_time + (timeMax * 1000)) {
+            count++;
+            SuspendAtExit threads[] = new SuspendAtExit[N_THREADS];
```

Comment:
Good catch!

As I'm re-reading this code, it occurs to me that I really don't need to
create the array of `N_THREADS` any more  since we are now time
based and we'll create a new Thread in each loop and run it through
the gauntlet until times runs out.

What do you think about changing `SuspendAtExit[N_THREADS]`
into just a single `SuspendAtExit`?